File: client/cs/src/Proton.Drive.Sdk/Telemetry/UploadBlockCount.cs
namespace Proton.Drive.Sdk.Telemetry;

// Numeric values must match the UploadBlockCount enum in proton.drive.sdk.proto.
public enum UploadBlockCount
{
    Unspecified = 0,
    Single = 1,
    Few = 2,
    Many = 3,
}

File: incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/extension/UploadBlockCount.kt
package me.proton.drive.sdk.extension

import me.proton.drive.sdk.telemetry.UploadBlockCount
import proton.drive.sdk.ProtonDriveSdk

fun ProtonDriveSdk.UploadBlockCount.toEnum() = when(this) {
    ProtonDriveSdk.UploadBlockCount.UPLOAD_BLOCK_COUNT_UNSPECIFIED -> UploadBlockCount.UNSPECIFIED
    ProtonDriveSdk.UploadBlockCount.UPLOAD_BLOCK_COUNT_SINGLE -> UploadBlockCount.SINGLE
    ProtonDriveSdk.UploadBlockCount.UPLOAD_BLOCK_COUNT_FEW -> UploadBlockCount.FEW
    ProtonDriveSdk.UploadBlockCount.UPLOAD_BLOCK_COUNT_MANY -> UploadBlockCount.MANY
    ProtonDriveSdk.UploadBlockCount.UNRECOGNIZED -> UploadBlockCount.UNRECOGNIZED
}
